yew/src/html/listener/registry.rs

```diff
@@ -0,0 +1,644 @@
+        }) {
+            Self::run_handlers(desc, event, target);
+        }
+    }
+
+    /// Handle a global event firing
+    fn handle(desc: EventDescriptor, event: Event) {
+        let target = match event
+            .target()
+            .map(|el| el.dyn_into::<web_sys::Element>().ok())
+            .flatten()
+        {
+            Some(el) => el,
+            None => return,
+        };
+
+        if desc.flags.has_set(DEFER) {
+            Registry::with(move |r| {
+                r.deferred.push_back((desc, event, target));
+                r.ensure_render_task();
+            });
+            return;
+        }
+
+        Self::run_handlers(desc, event, target);
+    }
+
+    fn run_handlers(desc: EventDescriptor, event: Event, target: web_sys::Element) {
+        let run_handler = |el: &web_sys::Element| {
+            if let Some(l) = LISTENER_ID_PROP
+                .with(|prop| js_sys::Reflect::get(el, &prop).ok())
+                .map(|v| v.dyn_into().ok())
+                .flatten()
+                .map(|num: js_sys::Number| {
+                    Registry::with(|r| {
+                        r.by_id
+                            .get(&(num.value_of() as u32))
+                            .map(|s| s.get(&desc))
+                            .flatten()
+                            .cloned()
+                    })
+                })
+                .flatten()
+            {
+                l.handle(event.clone());
+            }
+        };
+
+        run_handler(&target);
+
+        if Registry::with(|r| r.bubbling.contains(&desc)) {
+            let mut el = target;
+            loop {
+                el = match el.parent_element() {
+                    Some(el) => el,
+                    None => break,
+                };
+                run_handler(&el);
+            }
+        }
+    }
+}
+
+// Enable resetting between tests
+#[cfg(test)]
+impl Drop for Registry {
+    fn drop(&mut self) {
+        BODY.with(|body| {
+            for (kind, cl) in std::mem::take(&mut self.registered) {
+                AsRef::<web_sys::EventTarget>::as_ref(body)
+                    .remove_event_listener_with_callback(
+                        &kind.as_ref()[2..],
+                        cl.as_ref().unchecked_ref(),
+                    )
+                    .unwrap();
+            }
+        });
+    }
+}
+
+/// Register new event listeners to the element
+pub(crate) fn set_listeners(el: &Element, new: &mut Listeners) {
+    if let Listeners::Pending(ref mut pending) = new {
+        if pending.is_empty() {
+            return;
+        }
+
+        *new = Listeners::Registered(Registry::with(|reg| {
+            let id = reg.set_listener_id(el);
+            reg.register(id, std::mem::take(pending));
+            id
+        }));
+    }
+}
+
+/// Register new event listeners to the element, replacing old ones
+pub(crate) fn patch_listeners(mut new: &mut Listeners, old: &Listeners) {
```

Comment:
Looks like if the ancestor didn't have listeners (stuck in Pending) this will never apply new listeners

Reply:
Oh, true.